shell/common/shell.cc

```diff
@@ -1240,4 +1248,15 @@ Rasterizer::Screenshot Shell::Screenshot(
   return screenshot;
 }
 
+bool Shell::WaitForFirstFrame(fml::TimeDelta timeout) {
```

Comment:
Let's have a unit test for this in `shell_unittests.cc`.  Ideally, we could also use a pressure test that tries this a lot of times, with a randomized fake first frame rendering time that ranges from really fast (0ms) to really slow (>100ms).

Reply:
Done.